NVIDIA/deepops
# Implement a caching container registry for Slurm and k8s

This PR sets up a local Docker registry, with caching enabled for pulls from Docker Hub, and configures the nodes to pull their containers via this registry.

In the case of Kubernetes, registry mirrors will be deployed to each Kubernetes master.

In the case of Slurm, registry mirrors will be deployed to each Slurm controller.

## Test plan

1. Turn up a new cluster
1. On the registry host, check that there is a container `deepops-registry` running
1. Attempt to pull a container such as `ubuntu:latest` from any host in the cluster
1. Check the logs on the registry container and validate that the container was pulled via the registry container as a proxy.
